oauth.Dockerfile

```diff
@@ -9,13 +9,14 @@ COPY go.sum go.sum
 # and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-COPY static/ static/
 
 # Copy the go source
-COPY cmd/oauth cmd/oauth
+COPY cmd/ cmd/
 COPY api/ api/
-COPY oauth/ oauth/
 COPY pkg/ pkg/
+COPY oauth/ oauth/
+
+COPY static/ static/
```

Comment:
Isn't it better to keep this step before copying the source code? To utilize caching as the static folder is rarely (never) changed.

Reply:
yes, but cache is shared with operator dockerfile until line 17 and moving this up would break it and I see more benefit in shared cache with operator. 
However, I think there is place for improvement but tbh I'm not sure it's worth it. We can: 
 - move it just above `COPY oauth/ oauth/`, but I think it has very little benefit
 - move it in front of sources and do the same in operator as well, but we don't need it to build the operator
 - keep it like this patch proposes

WDYT?

Reply:
> yes, but cache is shared with operator dockerfile until line 17 and moving this up would break it and I see more benefit in shared cache with operator. 
> 
Ah yes good thinking.

> * move it just above `COPY oauth/ oauth/`, but I think it has very little benefit

I think I like this the most.

> * move it in front of sources and do the same in operator as well, but we don't need it to build the operator

This would probably be the most efficient but I don't like having the statics in the operator when we don't need them.

As you pointed out it's not much of a difference so feel free to pick what you find best.

Reply:
ok, I've move static above copy oauth